src/hotspot/share/code/nmethod.cpp

```diff
@@ -110,6 +110,11 @@
 
 #endif
 
+// Cast from int value to narrow type
+#define CHECKED_CAST(result, T, thing)      \
+  result = static_cast<T>(thing); \
+  assert(static_cast<int>(result) == thing, "failed: %d != %d", static_cast<int>(result), thing);
+
```

Comment:
I replaced `checked_cast<>()` with this macro because of next issues:
 - The existing assert points to `utilities/checkedCast.hpp` file where this method is located and not where failed cast. It does not help when it is used several times in one method (for example, in `nmethod()` constructors).
 - The existing assert does not print values

Reply:
I thought @kimbarrett had a draft PR to address the error reporting issue, but I can't seem to find it.  To solve the general problem, I think we need a version of vmassert() that takes `char* file, int lineno` as arguments, and a macro wrapper for checked_cast() that passes `__FILE__` and `__LINEN__` from the caller.

Reply:
Yes, it would be perfect separate RFE.